spotipy2/methods/playlist.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+from typing import List, Optional
+
+import spotipy2
+from spotipy2.types import Playlist, Track
+
+
+class PlaylistMethods:
+    async def get_playlist(
+        self: spotipy2.Spotify, playlist_id: str  # type: ignore
+    ) -> Playlist:
+        return Playlist.from_dict(
+            await self._get(f"playlists/{self.get_id(playlist_id)}")
+        )
+
+    async def get_playlist_tracks(
+        self: spotipy2.Spotify,  # type: ignore
+        playlist_id: str,
+        market: Optional[str] = None,
+        limit: Optional[int] = None,
+        offset: Optional[int] = None,
+    ) -> List[Track]:
+        limit = limit or 20
+        params = self.wrapper(market=market, limit=limit, offset=offset)
+
+        playlist_tracks = await self._get(
+            f"playlists/{self.get_id(playlist_id)}/tracks", params=params
+        )
+
+        tracks = [Track.from_dict(track["track"]) for track in playlist_tracks["items"]]
+
+        if offset:
```

Comment:
To maintain a certain order and same structure, the `get_playlist_tracks` method should do exactly one request, without extending the array to contain the entire tracklist.

For that, it's better to create a second generator method called `iter_playlist_tracks`, which makes multiple requests for [limit] objects only when it's needed and iterates through each one of the playlist tracks.